programs/stake/benches/stake.rs

```diff
@@ -0,0 +1,714 @@
+#![feature(test)]
+extern crate test;
+
+use {
+    bincode::serialize,
+    solana_feature_set::FeatureSet,
+    solana_program_runtime::invoke_context::mock_process_instruction,
+    solana_sdk::{
+        account::{create_account_shared_data_for_test, AccountSharedData, WritableAccount},
+        clock::{Clock, Epoch},
+        instruction::AccountMeta,
+        pubkey::Pubkey,
+        stake::{
+            instruction::{
+                self, AuthorizeCheckedWithSeedArgs, AuthorizeWithSeedArgs, LockupArgs,
+                LockupCheckedArgs, StakeInstruction,
+            },
+            stake_flags::StakeFlags,
+            state::{Authorized, Lockup, StakeAuthorize, StakeStateV2},
+        },
+        stake_history::StakeHistory,
+        sysvar::{
+            clock,
+            rent::{self, Rent},
+            stake_history,
+        },
+    },
+    solana_stake_program::{
+        stake_instruction,
+        stake_state::{Delegation, Meta, Stake},
+    },
+    solana_vote_program::vote_state::{self, VoteState, VoteStateVersions},
+    std::sync::Arc,
+    test::Bencher,
+};
+
+struct TestSetup {
+    feature_set: Arc<FeatureSet>,
+    stake_address: Pubkey,
+    stake_account: AccountSharedData,
+    transaction_accounts: Vec<(Pubkey, AccountSharedData)>,
+    instruction_accounts: Vec<AccountMeta>,
+}
+
+impl TestSetup {
+    fn new() -> Self {
+        let stake_account = AccountSharedData::new(
+            u64::MAX / 2,
+            StakeStateV2::size_of(),
+            &solana_stake_program::id(),
+        );
+        let stake_address = solana_sdk::pubkey::Pubkey::new_unique();
+        Self {
+            feature_set: Arc::new(FeatureSet::all_enabled()),
```

Comment:
Some of the `stake` instructions are behind feature gate, so enabling all for benches here